modules/core/core-api/src/main/java/com/enonic/xp/util/HashCode.java

```diff
@@ -0,0 +1,52 @@
+package com.enonic.xp.util;
+
+import com.enonic.xp.core.internal.HexCoder;
+
+public abstract class HashCode
+{
+    private HashCode()
+    {
+    }
+
+    public abstract String asString();
+
+    public String toString()
+    {
+        return asString();
+    }
+
+    public static HashCode fromLong( final long value )
+    {
+        return new LongHashCode( value );
+    }
+
+    private static final class LongHashCode
+        extends HashCode
+    {
+        private final long value;
+
+        LongHashCode( final long value )
+        {
+            this.value = value;
+        }
+
+        @Override
+        public String asString()
+        {
+            return HexCoder.toHex( value );
+        }
+
+        @Override
+        public boolean equals( final Object o )
```

Comment:
Is equality needed?

Reply:
Since it is now private and only implementation - there is some freedom in how it is implemented.
And generally good to have (to be able to collect into HashSet)